flyteidl/clients/go/coreutils/literals.go

```diff
@@ -559,12 +562,35 @@ func MakeLiteralForType(t *core.LiteralType, v interface{}) (*core.Literal, erro
 			strValue = fmt.Sprintf("%.0f", math.Trunc(f))
 		}
 		if newT.Simple == core.SimpleType_STRUCT {
+			// If the type is a STRUCT, we expect the input to be a complex object
```

Comment:
will this not break if the clients or otherside have the old view of struct? 
i.e, this seems backwards incompatible?
cc @EngHabu

Reply:
this will not break `JSON string` case, since this is not a string type.